File: fix_mainjs.js
// Script to fix main.js by removing cash register code
const fs = require('fs');

// Read the file
const content = fs.readFileSync('src/main.js', 'utf8');
const lines = content.split('\n');

console.log(`Original file has ${lines.length} lines`);

// Remove lines 247-310 (cash view block, 0-indexed so 246-309)
// Remove lines 196-198 (cash button, 0-indexed so 195-197)
const linesToRemove = new Set();

// Cash view block
for (let i = 246; i <= 309; i++) {
    linesToRemove.add(i);
}

// Cash button (adjust for already removed lines)
for (let i = 195; i <= 197; i++) {
    linesToRemove.add(i);
}

const newLines = lines.filter((line, index) => !linesToRemove.has(index));

console.log(`New file has ${newLines.length} lines`);
console.log(`Removed ${lines.length - newLines.length} lines`);

// Write back
fs.writeFileSync('src/main.js', newLines.join('\n'), 'utf8');

console.log('✅ Fixed main.js!');

File: fix_mainjs.py
#!/usr/bin/env python3
# Script to fix main.js by removing cash register code

import sys

# Read the file
with open('src/main.js', 'r', encoding='utf-8') as f:
    lines = f.readlines()

# Lines to remove (0-indexed, so subtract 1)
# Remove lines 247-310 (cash view block) and 196-198 (cash button)
skip_ranges = [
    (246, 310),  # Cash view commented block
    (195, 198)   # Cash button in sidebar
]

# Filter out the lines
new_lines = []
for i, line in enumerate(lines, start=1):
    should_skip = any(start <= i <= end for start, end in skip_ranges)
    if not should_skip:
        new_lines.append(line)

# Write back
with open('src/main.js', 'w', encoding='utf-8', newline='') as f:
    f.writelines(new_lines)

print(f"Fixed! Removed {len(lines) - len(new_lines)} lines")
print(f"Original: {len(lines)} lines, New: {len(new_lines)} lines")
